File: 2117-find-original-array-from-doubled-array/find-original-array-from-doubled-array.cpp
class Solution {
public:
    vector<int> findOriginalArray(vector<int>& changed) {

        if (changed.size() % 2 != 0)
            return {};
        sort(changed.begin(), changed.end());
        unordered_map<int, int> freq;
        for (int num : changed) {
            freq[num]++;
        }
        vector<int> original;
        for (int num : changed) {

            if (freq[num] == 0)
                continue;
            if (freq[2 * num] == 0)
                return {};
            original.push_back(num);
            freq[num]--;
            freq[2 * num]--;
        }

        return original;
    }
};

File: 2117-find-original-array-from-doubled-array/find-original-array-from-doubled-array.py
from collections import Counter
class Solution(object):
    def findOriginalArray(self, changed):
        if len(changed) % 2:
            return []

        count = Counter(changed)
        original = []
        for num in sorted(changed):
            if count[num] == 0:
                continue
            if count[2 * num] == 0:
                return []
            original.append(num)
            count[num] -= 1
            count[2 * num] -= 1

        return original